scipy/stats/_continuous_distns.py

```diff
@@ -10372,45 +10370,50 @@ def n_th_moment(n, a, b):
             Returns n-th moment. Defined only if n >= 0.
             Function cannot broadcast due to the loop over n
             """
-            pA, pB = self._pdf(np.asarray([a, b]), a, b)
-            probs = [pA, -pB]
+            ab = np.asarray([a, b])
+            pA, pB = self._pdf(ab, a, b)
+            probs = np.asarray([pA, -pB])
+            cond = probs.astype(bool)
```

Comment:
If you were tempted to write this as `cond = probs != 0`, I would agree with it.